README.md

```diff
@@ -42,19 +60,67 @@ health-checker [options]
 
 | Option | Description | Default 
 | ------ | ----------- | -------
-| `--port` | The port number on which a TCP connection will be attempted. Specify one or more times. | | 
 | `--listener` |  The IP address and port on which inbound HTTP connections will be accepted. | `0.0.0.0:5000`
+| `--config` | A YAML file containing checks which will be evaluated | `health-checks.yml`
 | `--log-level` | Set the log level to LEVEL. Must be one of: `panic`, `fatal`, `error,` `warning`, `info`, or `debug` | `info` 
 | `--help` | Show the help screen | | 
 | `--version` | Show the program's version | | 
 
-#### Example
+```
+health-checker --listener "0.0.0.0:6000" --config "my-checks.yml" --log-level "warning"
+```
 
-Run a listener on port 6000 that accepts all inbound HTTP connections for any URL. When the request is received,
-attempt to open TCP connections to port 5432 and 3306. If both succeed, return `HTTP 200 OK`. If any fails, return `HTTP
-504 Gateway Not Found`.
+#### Configuration
+
+##### tcp
+| Key       | Description 
+| --------- | ----------- 
+| `name`    | (Required) Name of the health check
+| `host`    | (Required) IP or hostname to check
+| `port`    | (Required) `port` to check on `host`
+| `timeout` | Timeout for health check in seconds - if not specified defaults to 5
+
+```yaml
+tcp:
+  - name: tcpService1
+    host: localhost
+    port: 5500
+```
 
+##### http
+| Key            | Description
+| -------------- | -----------
+| `name`         | (Required) Name of the health check
+| `host`         | (Required) IP or hostname to check
+| `port`         | (Required) `port` to check on `host`
+| `timeout`      | Timeout for health check in seconds - if not specified defaults to 5
+| `status_codes` | An array of status codes which should PASS health check
+| `body_regex`   | **Will not be checked if `status_codes` specified** - A string to search for in the body of the response, if found will pass health check
+
+```yaml
+http:
+  - name: httpService1
+    host: 127.0.0.1
+    port: 8080
+    success_codes: [200, 204, 301, 302]
+  - name: httpService2
+    host: 127.0.0.1
+    port: 8081
+    body_regex: "healthy"
 ```
-health-checker --listener "0.0.0.0:6000" --port 5432 --port 3306
+##### script
+| Key            | Description
+| -------------- | -----------
+| `name`         | (Required) Name of the health check
+| `script`       | (Required) Path to script to run - will PASS if it completes within 5s with a zero exit status
+| `timeout`      | Timeout for health check in seconds - if not specified defaults to 5
+
+```yaml
+script:
+  - name: scriptCheck1
+    script: /path/to/some/script.sh
```

Comment:
Seems like we should also support `args`, a list of arguments to pass to the script.